dimod/constrained.py

```diff
@@ -681,10 +685,21 @@ def add_variable(self, v: Variable, vartype: VartypeLike,
         Examples:
             >>> from dimod import ConstrainedQuadraticModel, Integer
             >>> cqm = ConstrainedQuadraticModel()
-            >>> cqm.add_variable('i', 'INTEGER')
+            >>> cqm.add_variable(INTEGER', 'i')
             'i'
 
         """
+        try:
+            vartype = as_vartype(vartype, extended=True)
+        except TypeError:
+            # in dimod<0.11 the argument order was v, vartype so let's allow that case
+            warnings.warn(
+                "Providing vartype and v in CQM.add_variable(v, vartype) order "
```

Comment:
```suggestion
                "Parameter order CQM.add_variable(v, vartype) "
```

Reply:
Maybe `Parameter order of CQM.add_variable(v, vartype)`? It feels like it's missing a word to me otherwise but I could be wrong

Reply:
We're not deprecating the existence of a parameter order in the function, just the particular order. We can make it more explicit, "Parameter order `v, vartype` in CQM.add_variable(v, vartype)"  if you like. 
I just found it not so easy to parse the current first line: the order comes late after the providing

Reply:
Ok, updated